### PR TITLE
Initialize storybook and responsible layout component

### PR DESCRIPTION
**Pull request by dmylian**
This is still a draft, the component does not fully work as intended.
Let's merge PR-1 and get back to this one

### CODE REVIEW

File: src/atoms/card/card.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+interface Props extends ComponentPropsWithoutRef<'div'> {
+  title: string,
+  key: string,
+}
+
+// https://stackoverflow.com/questions/54654303/using-a-forwardref-component-with-children-in-typescript
+// still don't know what's going on with the types
+
+export const Card = forwardRef<ComponentPropsWithoutRef<'div'>, PropsWithChildren<Props>>(
```
Comment (dmylian) [inline src/atoms/card/card.tsx]: todo: https://www.carlrippon.com/react-forwardref-typescript/

**Comment by dmylian** [comment]
- [x] compare with https://codesandbox.io/s/182lw48lo7?file=/src/components/GridItem.js, why both drag and resize work there?

File: src/components/atoms/card/card.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, {
+  forwardRef,
+  ComponentPropsWithoutRef,
+  PropsWithChildren,
+  ForwardedRef,
+} from "react";
+import styled from "styled-components";
+import Z from Y;
```
Comment (dmylian) [inline src/components/atoms/card/card.tsx]: Add an icon provider

File: src/components/atoms/card/card.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  PropsWithChildren<Props>
+>(({ style, className, ...props }, ref: any) => (
+  <CardRoot
+    style={{ ...style }}
+    className={className}
+    ref={ref}
+    key={props.key}
+    {...props}
+  >
+    <Headline>{props.title}</Headline>
```
Comment (dmylian) [inline src/components/atoms/card/card.tsx:47]: Add a draggable control

File: src/components/atoms/resize-handle/ResizeHandle.tsx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import React from "react";
+import X from Y;
```
Comment (dmylian) [inline src/components/atoms/resize-handle/ResizeHandle.tsx]: Add an icon provider

File: src/components/atoms/resize-handle/ResizeHandle.tsx [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+
+export const Handle = () => {};
```
Comment (dmylian) [inline src/components/atoms/resize-handle/ResizeHandle.tsx]: Implement a resize handle, consumable by a https://github.com/react-grid-layout/react-grid-layout#custom-child-components-and-draggable-handles

File: src/components/atoms/responsive-grid/responsive-grid.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+
+import { ResponsiveGrid } from "./responsive-grid";
+import { Card } from "../card/card";
+
+export default {
+  title: "Example/ResponsiveGrid",
+  component: ResponsiveGrid,
+  argTypes: {},
+} as ComponentMeta<typeof ResponsiveGrid>;
```
Comment (dmylian) [inline src/components/atoms/responsive-grid/responsive-grid.stories.tsx:11]: Storybook story should be working :)

File: src/components/atoms/responsive-grid/responsive-grid.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import "./styles.css";
+
+interface Props {
+  children(l: any, index: number): ReactNode;
+}
+
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+export const ResponsiveGrid = (props: ResponsiveGridLayoutProps & Props) => {
+  return null;
```
Comment (dmylian) [inline src/components/atoms/responsive-grid/responsive-grid.tsx]: Add the implementation

File: src/components/atoms/responsive-grid/styles.css [hunk 8]
```diff
@@ -0,0 +1,5 @@
+.react-grid-item.react-grid-placeholder {
+  /*background: #efefef;*/
+  /*border: 2px dashed grey;*/
+  border-radius: 4px;
+}
```
Comment (dmylian) [inline src/components/atoms/responsive-grid/styles.css]: You can play with the grid styling here. Maybe change the bg color or leave as is

File: src/react-app-env.d.ts [hunk 9]
```diff
@@ -1 +1,81 @@
 /// <reference types="react-scripts" />
+
+declare module 'react-grid-layout' {
```
Comment (dmylian) [inline src/react-app-env.d.ts:3]: I haven't found ts typings for the module, so decided to add a naive ad-hoc solution, check out how types and interfaces describe the existing library contract

**Comment by dmylian** [comment]
- [x] implement responsive grid
- [x] add custom resize handle
- [x] add custom drag handle
- [x] make sure story works
- [ ] maybe play with 'drag from outside' feature as it will be required by the app

File: src/components/atoms/responsive-grid/styles.css (hunk 8 above)
Comment (chooidos) [reply src/components/atoms/responsive-grid/styles.css]: we have CardRoot styled component, do we need this styling here?

Comment (dmylian) [reply src/components/atoms/responsive-grid/styles.css]: If you want to change the color of the drop-zone placeholder while dragging an element, we do. CardRoot's style will not apply to the drag-related elements

**Comment by dmylian** [comment]
@chooidos please merge

**Comment by chooidos** [comment]
> @chooidos please merge

@dmylian  resize doesn't work, fixing

**Pull request merged**